### DISCUSSION TITLE
Assign more than hotkey to the same action

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4018] [Q&A]
I was wondering if it was possible to be able to assign more than one hotkey to the same action.
For example:
`Notebook Tab - Switch to right document = Ctrl+PageDown`
What if user also wants to assign another hotkey for the same action, only allows for one.

Thank You

**Comment by elextr** [comment]
Unfortunately no.

The Geany keybinding dialog has a fixed list of actions that can each only be assigned one keycode.

The dialog has a number of issues as well as this, but no one has attempted to recode it. So it is unlikely that it will be changed.

**Comment by Skif-off** [comment]
I'm not sure, but it seems you can try a workaround with the [GeanyLua](https://github.com/geany/geany-plugins/tree/master/geanylua) plugin (see `docs/geanylua-keys.html`). Something like `SwitchToRightDocument.lua` (with your example):
```lua
geany.keycmd("NOTEBOOK_SWITCHTABRIGHT")
```

**Comment by advice2020** [comment]
@elextr
Ahh that is unfortunate, thank you.
You mentioned that the Preferences dialog window has a number of issues.
Probably better that I do not know what they are :) ..... but
Are they more coding based issues that only a developer would understand or are there any more layout / feature request based kind of issues?
If they are layout/feature based let me know and maybe I could at least type out some posts for the Ideas section just in case future developers are interested in making the updates.
Up to you.
I will make a request post in Idea section for this just in case.

Thank You

**Comment by advice2020** [comment]
@Skif-off
Sorry I did not realize you responded, thank you for this.
Also took me a little while to be able to test this, even though I knew of the GeanyLua plugin and its use (because a different plugin I was looking into required it) because Geany is used for creating scripts, I still was viewing that plugin as if it added help/tools for scripting in Lua, rather than what it actually does, providing Lua scripts to be used by Geany itself.  Hopefully that makes sense :)
Anyways, good idea I was able to try this and it works great.
As mentioned above, I think I am still going to put in a request for this, but as @elextr mentioned this probably is not going to be added, so thank you for providing this solution.
Can I just ask you one question, what you provided is the main keybinding I was looking for at the moment....
`geany.keycmd("NOTEBOOK_SWITCHTABLEFT")`
`geany.keycmd("NOTEBOOK_SWITCHTABRIGHT")`
but where did you find the `"NOTEBOOK_SWITCHTABRIGHT"` actions name?
Do you know of a list of actions somewhere, because I might want to assign other actions to more than one keybinding.

Thank You again

**Comment by Skif-off** [comment]
> where did you find the `"NOTEBOOK_SWITCHTABRIGHT"` actions name?

In the plugin's source code, [glspi_keycmd.h](https://github.com/geany/geany-plugins/blob/master/geanylua/glspi_keycmd.h) is mentioned in the description of the plugin's functions (see `docs/geanylua-ref.html`).

**Comment by advice2020** [comment]
@Skif-off
Ahh I am glad I asked :)
Thank You.
